src/routes/profile.js

```diff
@@ -386,15 +398,28 @@ router.post('/:profileUsername', checkProfileOwnership, (req, res) => {
 router.get('/:profileUsername', (req, res) => {
   User.findOne(
     { usernameLower: req.params.profileUsername.toLowerCase() },
-    (err, foundUser) => {
+    async (err, foundUser) => {
       if (err) {
         console.log(err);
       } else {
+        //get the parameters of user that to render the profile page
+        const currentRanking = await RankData.findById(foundUser.currentRanking).exec();
+        const pastRankings = [];
+        const currentSeasonNumber = await getSeasonNumber();
+        //get the past rankings of the user and reverse the order
+        for (const element of foundUser.pastRankings) {
+          const pastRanking = await RankData.findById(element).exec();
+          pastRankings.unshift(pastRanking);
+        }
+
         res.render('profile/profile', {
           userData: foundUser,
-          seasonNumber: seasonNumber.getSeasonNumber(),
+          seasonNumber: currentSeasonNumber,
+          currentRanking: currentRanking,
+          pastRankings: pastRankings,
```

Comment:
Can we reduce this a bit and simply pass in a list of rankings? Each ranking should have a season number on it. You can then assume that the last ranking is the most recent one.

Reply:
You can get rid of `getSeasonNumber()` as well in this case.